rust/automerge/tests/test.rs

```diff
@@ -1412,6 +1412,31 @@ fn fuzz_crashers() {
     }
 }
 
+#[test]
+fn overlong_leb() {
+    // the value metadata says "2", but the LEB is only 1-byte long and there's an extra 0
+    let input = vec![
+        133, 111, 74, 131, 101, 189, 58, 107, 0, 110, 1, 16, 189, 164, 138, 64, 231, 24, 77, 199,
+        169, 158, 106, 229, 216, 158, 46, 212, 1, 202, 20, 126, 211, 151, 175, 73, 116, 227, 106,
+        198, 46, 28, 245, 111, 85, 96, 236, 50, 60, 255, 184, 1, 237, 26, 202, 247, 135, 165, 111,
+        117, 39, 6, 1, 2, 3, 2, 19, 2, 35, 2, 64, 2, 86, 2, 8, 21, 3, 33, 2, 35, 2, 52, 1, 66, 2,
+        86, 2, 87, 2, 128, 1, 2, 127, 0, 127, 1, 127, 1, 127, 0, 127, 0, 127, 7, 127, 1, 97, 127,
+        0, 127, 1, 1, 127, 1, 127, 40, /* */ 28, 0, /* */ 127, 0,
+    ];
```

Comment:
I'm also happy to put these fixtures into files instead of inline if that's better.